tools/js2c.py

```diff
@@ -214,7 +217,16 @@ def JS2C(source, target):
     lines = ReadFile(str(s))
     lines = ExpandConstants(lines, consts)
     lines = ExpandMacros(lines, macros)
-    data = ToCString(lines)
+
+    # Treat non-ASCII as UTF-8 and convert it to UTF-16.
+    if any(ord(c) > 127 for c in lines):
+      ctype = 'uint16_t'
+      data = map(ord, lines.decode('utf-8').encode('utf-16be'))
```

Comment:
The big-endian encoding is intentional, the line below stitches the individual octets together again into uint16s.  (`.encode()` returns a byte string in case you're wondering.)